karapace/master_coordinator.py

```diff
@@ -64,7 +66,7 @@ def _perform_assignment(self, leader_id, protocol, members):
         fallback_urls = {}
         for member_id, member_data in members:
             member_identity = json.loads(member_data.decode("utf8"))
-            if member_identity["master_eligibility"] is True:
+            if member_identity["master_eligibility"] is True and member_identity.get("ready", False) is True:
```

Comment:
Note the default `False` for ready. Effectively this means that Karapace that does not provide `ready` is never allowed to become a primary as the state of those is unknown. This is important to know in scenarios where rolling update to newer version is done.

Reply:
After latest refactoring the `member_identity` struct no longer contains `ready` flag at all...